### PR TITLE
Adds turret bases

### PR DESCRIPTION
**Pull request by Dorsisdwarf**
Pop-up turrets now have a base underneath them because it was weird as fuck that they popped out from under clamshell armoured doors... which were just sitting on top of the floor?

https://gyazo.com/9b3be51d9e56c0ec58bee435c7bd0c48

### CODE REVIEW

**Comment by RemieRichards** [comment]
this could all just be an underlay...

**Comment by GunHog** [comment]
Image is broken for me

**Comment by RemieRichards** [comment]
@GunHog https://i.gyazo.com/9b3be51d9e56c0ec58bee435c7bd0c48.mp4
it's a video, it doesn't embed

**Comment by GunHog** [comment]
Oh, neato

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -920,3 +929,42 @@
...
+
+/obj/structure/turretbase/attackby(obj/item/I, mob/user, params)
+	if(!parent_turret)
+		if(istype(I,/obj/item/stack/tile))
+			var/obj/item/stack/tile/T = I
+			if(!T.use(1))
+				return
+			else
+				playsound(src, 'sound/weapons/Genhit.ogg', 50, 1)
+				src.Destroy()
```
Comment (ChangelingRain) [inline code/game/machinery/porta_turret/portable_turret.dm]: uh

Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: HGGNNGNGGGNGNGHHHHHHHH

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 2]
```diff
@@ -87,6 +90,8 @@
 	if(has_cover)
 		cover = new /obj/machinery/porta_turret_cover(loc)
 		cover.parent_turret = src
+		underbase = new /obj/structure/turretbase(loc)
```
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: remove the `/obj/structure/turretbase` part

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 3]
```diff
@@ -920,3 +929,42 @@
...
+
+//---------------turret bases -------------
+
+/obj/structure/turretbase
+	name = "recessed turret bay"
+	desc = "A wonder of modern science, the BS-43 Portable Turret Hole allows for easy portability of pop-up shielded turrets."
+	icon = 'icons/obj/turrets.dmi'
+	flags = INDESTRUCTIBLE
+	icon_state = "basedark"
+	anchored = 1
```
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: TRUE

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 4]
```diff
@@ -920,3 +929,42 @@
...
+
+/obj/structure/turretbase/attack_alien()
+	return
+
+/obj/structure/turretbase/attack_hulk()
+	return
+
+/obj/structure/turretbase/attack_animal()
+	return
+
```
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: Need to add singularity_pull override

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 5]
```diff
@@ -920,3 +929,42 @@
...
+/obj/structure/turretbase
+	name = "recessed turret bay"
+	desc = "A wonder of modern science, the BS-43 Portable Turret Hole allows for easy portability of pop-up shielded turrets."
+	icon = 'icons/obj/turrets.dmi'
+	flags = INDESTRUCTIBLE
+	icon_state = "basedark"
+	anchored = 1
+	layer = LOW_OBJ_LAYER
+	var/obj/machinery/porta_turret/parent_turret = null
```
Comment (Cyberboss) [inline code/game/machinery/porta_turret/portable_turret.dm]: Remove `= null`, Set this to null in Destroy

**Review by Cyberboss — CHANGES_REQUESTED** [review]
I whole heartedly agree with @RemieRichards , just use an underlay

**Comment by Dorsisdwarf** [comment]
I'll switch this to an underlay once I unfuck my git

**Comment by optimumtact** [comment]
FOR THE LOVE OF CHRIST STOP USING GYAZO, USE IMGUR OR SIGN UP FOR A https://file.house/ account

They both don't require javascript and don't show as many ads

Sharex and android app integration available

**Comment by Dorsisdwarf** [comment]
this is the first time I've used gyazo you dingus, also its the only method of screencapping a gif that I know off-hand

**Comment by optimumtact** [comment]
Just record a video you meme, gif is the most backwards format

**Comment by Dorsisdwarf** [comment]
sorry for not having a billion gigabytes for fraps just so I can show you nerds a non-static image

**Comment by optimumtact** [comment]
IT TAKES LESS FUCKING SPACE TO SAVE AS A VIDEO THAN A GIFFFF WHAT THE FUCK ARE YOU SAYINNNNNNNNNNNNNNNNNG

http://www.sohamkamani.com/blog/2016/04/09/stop-using-gifs/

**Comment by Dorsisdwarf** [comment]
All I know of video recording is fraps. I have literal hard drives filled with nothing but hundreds of gigabyte-large fraps videos

**Comment by PKPenguin321** [comment]
Get sharex

**Comment by optimumtact** [comment]
@Dorsisdwarf fraps records them in raw, then you encode them into something like .ogg or mpv etc

**Pull request closed without merging**